app/models/merged_search_service.rb

```diff
@@ -0,0 +1,319 @@
+require 'digest'
+
+# Orchestrates merged "all" tab searches across Primo and TIMDEX.
+#
+# Handles parallel fetches, per-query totals caching, pagination calculation via
+# `MergedSearchPaginator`, and assembly of a controller-friendly response hash.
+class MergedSearchService
+  # Time to live value for cache expiration.
+  TTL = 12.hours
+
+  # Initialize a new MergedSearchService.
+  #
+  # The service supports dependency injection for the two fetchers to keep
+  # calling code simple and to make testing deterministic.
+  #
+  # Usage patterns:
+  # - Default: callers omit fetcher arguments and the service uses `Rails.cache`
```

Comment:
It's still a bit unclear to me if we ever use the default path or if we are always injecting...